### PR TITLE
Feature3 merchant dashboard stats fav customers

### LINKED ISSUE #5: feature/3-merchant-dashboard-stats-fav-customers
3. Merchant Dashboard Statistics - Favorite Customers

As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the names of the top 5 customers
who have conducted the largest number of successful transactions with my merchant
And next to each customer name I see the number of successful transactions they have
conducted with my merchant

### PR DESCRIPTION
**Pull request by jpnelson85**
## Checklist before requesting a review
- [x ] Verified Render deploy is functional
- [x ] Pulled/merged main 
- [x ] Tests for the changes have been added into spec files
- [x ] Branch name follows our guidelines 
- [ x] All tests running clean
- [ x] Ran this in Rails S  - features are working properly
- [ x] Test coverage is at 100% for models and features

## Bugs, Issues, etc:
 - [ ] check if there are known bugs or issue

### CODE REVIEW

File: spec/features/merchants/show_spec.rb [hunk 1]
```diff
@@ -30,4 +30,72 @@
...
+      end
+    end
+
+    it "displays number of successful transactions for top 5 customers" do
+      visit "/merchants/#{merchant_1.id}/dashboard"
+
+      within("#top5") do
+      save_and_open_page
+      expect("#{customer_1.successful_transactions_count} successful transactions").to appear_before("#{customer_2.successful_transactions_count} successful transactions")
+      expect("#{customer_2.successful_transactions_count} successful transactions").to appear_before("#{customer_3.successful_transactions_count} successful transactions")      # expect(customer_3.successful_transactions_count).to appear_before(customer_4.successful_transactions_count)
```
Comment (AlexKiselich) [inline spec/features/merchants/show_spec.rb]: Looks like an # expect statement was left in at the end of this line

File: spec/features/merchants/show_spec.rb [hunk 2]
```diff
@@ -30,4 +30,72 @@
...
+      expect(customer_4.first_name).to appear_before(customer_5.first_name)
+      expect(page).to_not have_content(customer_6.first_name)
+      end
+    end
+
+    it "displays number of successful transactions for top 5 customers" do
+      visit "/merchants/#{merchant_1.id}/dashboard"
+
+      within("#top5") do
+      save_and_open_page
```
Comment (AlexKiselich) [inline spec/features/merchants/show_spec.rb]: save and open page

**Review by AlexKiselich — CHANGES_REQUESTED** [review]
Just some small stuff

File: spec/features/merchants/show_spec.rb [hunk 3]
```diff
@@ -30,4 +30,72 @@
...
+        create(:transaction, result: false, invoice: invoice)
+      end
+    end
+    
+    it "display names of top 5 customers with largest successful transactions " do
+    
+      visit "/merchants/#{merchant_1.id}/dashboard"
+
+      within("#top5") do
+      expect(customer_1.first_name).to appear_before(customer_2.first_name)
```
Comment (aplasket) [inline spec/features/merchants/show_spec.rb]: tiny thing that you don't have to change but can just note for future -- for within blocks, it's helpful to indent the expect statements so we can visually know to expect those assertions within that specific block in the view.

**Review by aplasket — APPROVED** [review]
other than what Alex has already mentioned, I only had 1 other small thing noted. Great job on this

Also note for future, the branch name should be `feature/3-merchant-dash....etc`; looks like it's missing the `/`

**Review by AlexKiselich — APPROVED** [review]

**Pull request merged**